lib/pricegrabber/response.rb

```diff
@@ -9,10 +9,10 @@ class Response
     # @param wants [Array<String>] Attributes from the response that should be made available within this response
     def initialize(response, wants)
       @status = response.code
-      resp_hash = Hash.from_xml(response.body)
-      @error = resp_hash["document"]["error"]
       @results = []
-      [resp_hash["document"]["product"]].flatten.each do |product|
+      return if @status != 200
+      resp_hash = Hash.from_xml(response.body)
+      [resp_hash["ProductResponse"]["Offers"]["Offer"]].flatten.each do |product|
```

Comment:
Looking at the diff we have locked this down to offers only. That is fine for our use, but is good reason not to try and merge it up to main code base.